src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -1498,6 +1537,43 @@ static Value builtinArrayPush(ExecutionState& state, Value thisValue, size_t arg
     return Value(n);
 }
 
+// https://www.ecma-international.org/ecma-262/10.0/#sec-array.prototype.flat
+// Array.prototype.flat( [ depth ] )
+static Value builtinArrayFlat(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(O, Array, flat);
+    int64_t sourceLen = O->lengthES6(state);
+    double depthNum = 1;
+    if (argc > 0) {
+        depthNum = argv[0].toInteger(state);
+    }
+    // if (std::isinf(depthNum)) {
+    //   depthNum = std::numeric_limits<int64_t>::max();
+    // }
```

Comment:
According to the spec,
> If depth is not undefined, then
    Set depthNum to ? ToInteger(depth).

We should check undefined value of `argv[0]` too.
And the last thing, please remove the comments which seems not related to this builtin function.